lib/Dialect/Arc/ModelInfo.cpp

```diff
@@ -116,7 +116,8 @@ LogicalResult circt::arc::collectModels(mlir::ModuleOp module,
     llvm::sort(states, [](auto &a, auto &b) { return a.offset < b.offset; });
 
     models.emplace_back(std::string(modelOp.getName()), storageType.getSize(),
-                        std::move(states), modelOp.getInitialFnAttr());
+                        std::move(states), modelOp.getInitialFnAttr(),
+                        modelOp.getFinalFnAttr());
```

Comment:
Do we want to add the finalizer function name to the JSON such that an external test bench can pick it up rather than guessing or hardcoding the function name?

Reply:
Great idea! I've added that :+1: